lib/bundler/runtime.rb

```diff
@@ -37,7 +37,17 @@ def setup(*groups)
 
         Bundler.rubygems.mark_loaded(spec)
         load_paths = spec.load_paths.reject {|path| $LOAD_PATH.include?(path) }
-        $LOAD_PATH.unshift(*load_paths)
+
+        # See Gem::Specification#add_self_to_load_path (since RubyGems 1.8)
+        insert_index = Gem.load_path_insert_index
```

Comment:
Found it [here](https://github.com/bundler/bundler/blob/master/lib/bundler/rubygems_integration.rb)

Comparing [this chunk of the PR](https://github.com/Elffers/bundler/blob/hhh_fix_ruby-I/lib/bundler/runtime.rb#L39-L50) to [RubyGems' implementation](https://github.com/rubygems/rubygems/blob/master/lib/rubygems/specification.rb#L1539-L1554) how do you propose we handle the reject on L39 from bundler? It was introduced by @69acfd88 by @indirect with no tests so I'm unsure what fault it addresses.